source/opt/pch.h

```diff
@@ -0,0 +1,19 @@
+
+#include <algorithm>
+#include <functional>
+#include <map>
+#include <queue>
+#include <unordered_map>
+#include <unordered_set>
+#include <stack>
+#include <utility>
+#include <vector>
```

Comment:
This list seems arbitrary and likely to fall out of date quickly. How long is the tools build time? Is that building everything including the tests or just the source code?

Reply:
precompiled header includes don't have to be perfect, generally as long as you catch the top few headers (which often include most of the headers in the project) it's fine. We've used PCH in the NVIDIA driver build for many years and almost never update the PCH headers.

The total build time for the spirv-tools solution on my system went from about 5 minutes 10 seconds to about 4 minutes 30 seconds. Each spirv-tools-opt file went from around 3 seconds to 1 second.

There appears to be some problem with the PCH in the windows CI builds, which is surprising. I guess it must be Ninja-specific because I can't reproduce it with visual studio.